/createMerchantTransaction Integration

ag3722 opened the pull request:
- Integrated /createMerchantTransaction into QR scanning/Award points page

ag3722 committed:
/createMerchantTransaction Integration

- Integrated /createMerchantTransaction into QR scanning/Award points page

SaifA21 commented on app/(auth)/awardpoints.tsx:
```diff
@@ -10,64 +10,128 @@ import {
+    }
+
+    const customerId = userInfo?.id;
+    const pointsEquivalent = Math.round(Number(subtotal));
```
We should use Math.floor since we only award a point for every complete dollar spent.

SaifA21 reviewed (changes requested):
Small change request also, API call is failing as customerId is not being passed in correctly.

![image](https://github.com/user-attachments/assets/788b4437-8e70-4f30-8641-41475ef8f976)

ag3722 committed:
Fix API call

- Fixed API call to pass userInfo as prop when navigating from QR scan page
- Fixed point rounding

SaifA21 reviewed (approved):
LGTM 🚀
